hack/golangci-strict.yaml

```diff
@@ -0,0 +1,54 @@
+# This file configures checks that all new code for Kubernetes is meant to
+# pass, in contrast to .golangci.yaml which defines checks that also the
+# existing code passes.
+
+run:
+  timeout: 30m
+  skip-files:
+    - "^zz_generated.*"
+
+issues:
+  max-same-issues: 0
+  # Excluding configuration per-path, per-linter, per-text and per-source
+  exclude-rules:
+    # exclude ineffassing linter for generated files for conversion
+    - path: conversion\.go
+      linters:
+        - ineffassign
+    - path: .*_test.go
+      linters:
+        - errcheck
```

Comment:
I disabled errcheck for test code because that was by far the most common warning for recent code.

But I am not sure whether this is the right decision and am leaning more towards warning about that also in test code: if something fails and the error doesn't cause a test failure, then understanding why the test fails later on will just be harder.